### PR TITLE
modernization TOF workflows

### PR DESCRIPTION
**Pull request by noferini**
@davidrohr

### CODE REVIEW

**Comment by noferini** [comment]
Hi @davidrohr ,
I implemented all the six points in the list for modernization of the code (I hope). No QC from consul here


1) I moved to add_W removing the pipeline at the end. It is not clear to me if pipeline is properly added before  the last workflow "o2-dpl-run". I assumed yes since I found the same in the exampe. Is it correct?
2) Did I interpret correctly your requirement for calibration workflows?

File: DATA/production/calib/tof-diagn-aggregator.sh [hunk 1]
```diff
@@ -5,14 +5,25 @@ source common/setenv.sh
...
-WORKFLOW+="o2-calibration-tof-diagnostic-workflow --tf-per-slot 25000 --max-delay 1 $ARGS_ALL | "
-WORKFLOW+="o2-calibration-ccdb-populator-workflow $ARGS_ALL --configKeyValues \"$ARGS_ALL_CONFIG\" --ccdb-path=\"http://ccdb-test.cern.ch:8080\" | "
-WORKFLOW+="o2-dpl-run $ARGS_ALL $GLOBALDPLOPT"
+WORKFLOW=
+add_W o2-dpl-raw-proxy "$ARGS_ALL --proxy-name tof-diagn-input-proxy --dataspec ${PROXY_INSPEC} --network-interface ib0 --channel-config \"name=tof-diagn-input-proxy,method=bind,type=pull,rateLogging=0,transport=zeromq\" "
+add_W o2-calibration-tof-diagnostic-workflow "--tf-per-slot 25000 --max-delay 1 ${ARGS_ALL}"
+
+if [[ $RUNTYPE == "SYNTHETIC" || "${GEN_TOPO_DEPLOYMENT_TYPE:-}" == "ALICE_STAGING" ]]; then
+    CCDB_POPULATOR_UPLOAD_PATH="http://ccdb-test.cern.ch:8080"
+fi
```
Comment (martenole) [inline DATA/production/calib/tof-diagn-aggregator.sh:19]: please remove erroneous `fi`

File: DATA/production/calib/tof-diagn-aggregator.sh [hunk 2]
```diff
@@ -5,14 +5,25 @@ source common/setenv.sh
 # ---------------------------------------------------------------------------------------------------------------------
 # Set general arguments
 source common/getCommonArgs.sh
+source common/gen_topo_helper_functions.sh
 ARGS_ALL_CONFIG="keyval.input_dir=$FILEWORKDIR;keyval.output_dir=/dev/null;$ALL_EXTRA_CONFIG"
```
Comment (martenole) [inline DATA/production/calib/tof-diagn-aggregator.sh:9]: the line `ARGS_ALL_CONFIG=...` is obsolete and can be removed (see getCommonArgs.sh)

File: DATA/production/calib/tof-standalone-cosmic-reco-time-calib.sh [hunk 3]
```diff
@@ -5,17 +5,19 @@ source common/setenv.sh
 # ---------------------------------------------------------------------------------------------------------------------
 # Set general arguments
 source common/getCommonArgs.sh
+source common/gen_topo_helper_functions.sh
 ARGS_ALL_CONFIG="keyval.input_dir=$FILEWORKDIR;keyval.output_dir=/dev/null;$ALL_EXTRA_CONFIG"
```
Comment (martenole) [inline DATA/production/calib/tof-standalone-cosmic-reco-time-calib.sh:9]: same as above

File: DATA/production/calib/tof-time-calib-aggregator.sh [hunk 4]
```diff
@@ -5,14 +5,24 @@ source common/setenv.sh
...
-WORKFLOW+="o2-calibration-tof-calib-workflow --cosmics --do-channel-offset --min-entries 1000 $ARGS_ALL | "
-WORKFLOW+="o2-calibration-ccdb-populator-workflow $ARGS_ALL --configKeyValues \"$ARGS_ALL_CONFIG\" --ccdb-path=\"http://ccdb-test.cern.ch:8080\" | "
-WORKFLOW+="o2-dpl-run $ARGS_ALL $GLOBALDPLOPT"
+WORKFLOW=
+add_W o2-dpl-raw-proxy "$ARGS_ALL --proxy-name tof-time-calib-input-proxy --dataspec ${PROXY_INSPEC} --network-interface ib0 --channel-config \"name=tof-time-calib-input-proxy,method=bind,type=pull,rateLogging=0,transport=zeromq\" "
+add_W o2-calibration-tof-calib-workflow "--cosmics --do-channel-offset --min-entries 1000 ${ARGS_ALL} "
+
+if [[ $RUNTYPE == "SYNTHETIC" || "${GEN_TOPO_DEPLOYMENT_TYPE:-}" == "ALICE_STAGING" ]]; then
+    CCDB_POPULATOR_UPLOAD_PATH="http://ccdb-test.cern.ch:8080"
+fi
```
Comment (martenole) [inline DATA/production/calib/tof-time-calib-aggregator.sh:19]: as above

File: DATA/production/calib/tof-standalone-reco.sh [hunk 5]
```diff
@@ -5,17 +5,19 @@ source common/setenv.sh
 # ---------------------------------------------------------------------------------------------------------------------
 # Set general arguments
 source common/getCommonArgs.sh
+source common/gen_topo_helper_functions.sh
 ARGS_ALL_CONFIG="keyval.input_dir=$FILEWORKDIR;keyval.output_dir=/dev/null;$ALL_EXTRA_CONFIG"
```
Comment (martenole) [inline DATA/production/calib/tof-standalone-reco.sh:9]: as above

File: DATA/production/calib/tof-time-calib-aggregator.sh [hunk 6]
```diff
@@ -5,14 +5,24 @@ source common/setenv.sh
 # ---------------------------------------------------------------------------------------------------------------------
 # Set general arguments
 source common/getCommonArgs.sh
+source common/gen_topo_helper_functions.sh
 ARGS_ALL_CONFIG="keyval.input_dir=$FILEWORKDIR;keyval.output_dir=/dev/null;$ALL_EXTRA_CONFIG"
```
Comment (martenole) [inline DATA/production/calib/tof-time-calib-aggregator.sh:9]: as above

**Review by martenole — CHANGES_REQUESTED** [review]
looks good, I just have some minor comments

File: DATA/production/calib/tof-diagn-aggregator.sh (hunk 1 above)
Comment (noferini) [reply DATA/production/calib/tof-diagn-aggregator.sh]: ops. Done. Thanks

File: DATA/production/calib/tof-diagn-aggregator.sh (hunk 2 above)
Comment (noferini) [reply DATA/production/calib/tof-diagn-aggregator.sh]: done

File: DATA/production/calib/tof-time-calib-aggregator.sh (hunk 4 above)
Comment (noferini) [reply DATA/production/calib/tof-time-calib-aggregator.sh]: done

File: DATA/production/calib/tof-standalone-reco.sh (hunk 5 above)
Comment (noferini) [reply DATA/production/calib/tof-standalone-reco.sh]: done

File: DATA/production/calib/tof-time-calib-aggregator.sh (hunk 6 above)
Comment (noferini) [reply DATA/production/calib/tof-time-calib-aggregator.sh]: done

**Review by martenole — APPROVED** [review]
Thanks, looks good! Concerning your questions:
1. yes, the pipe is added automatically by the add_W function, so your implementation is OK
2. also the CCDB upload path logic is correct

**Pull request merged**